homeassistant/components/niu/sensor.py

```diff
@@ -0,0 +1,258 @@
+"""Platform for sensor integration."""
+from datetime import timedelta
+import logging
+
+import async_timeout
+from niu import NiuAPIException
+
+from homeassistant.const import (
+    DEVICE_CLASS_BATTERY,
+    DEVICE_CLASS_TEMPERATURE,
+    LENGTH_KILOMETERS,
+    PERCENTAGE,
+    TEMP_CELSIUS,
+)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config, async_add_entities):
+    """Set up the sensor platform."""
+    entities = []
+
+    # Get Niu Api Object
+    niu_api = hass.data[DOMAIN][config.entry_id]
+
+    async def async_update_data():
+        """Fetch data from API endpoint.
+
+        This is the place to pre-process the data to lookup tables
+        so entities can quickly look up their data.
+        """
+        try:
+            # Note: asyncio.TimeoutError and aiohttp.ClientError are already
+            # handled by the data update coordinator.
+            async with async_timeout.timeout(10):
+                await niu_api.update_vehicles()
+                return niu_api.get_vehicles()
+        except NiuAPIException as ex:
+            raise UpdateFailed(f"Error communicating with API: {ex}") from ex
+
+    # Create coordinator that is used to update all the data
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.
+        name="sensor",
+        update_method=async_update_data,
+        # Polling interval. Will only be polled if there are subscribers.
+        update_interval=timedelta(seconds=60),
+    )
+
+    # Fetch initial data so we have data when entities subscribe / Get all scooters from Niu API
+    await coordinator.async_refresh()
+
+    # Get all scooters from Niu API and add the entities to the entity list
+    for serno, vehicle in coordinator.data.items():
+        _LOGGER.info("Found Niu vehicle: %s", vehicle.name)
```

Comment:
Home Assistant will already log the creation of entities in debug mode. Maybe remove this or make it debug level?